pkg/cmd/grafana-server/commands/diagnostics.go

```diff
@@ -90,15 +102,20 @@ func (td *tracingDiagnostics) overrideWithEnv() error {
 	return nil
 }
 
-func setupProfiling(profile bool, profileAddr string, profilePort uint64) error {
-	profileDiagnostics := newProfilingDiagnostics(profile, profileAddr, profilePort)
+func setupProfiling(profile bool, profileAddr string, profilePort uint64, profileContention bool) error {
+	profileDiagnostics := newProfilingDiagnostics(profile, profileAddr, profilePort, profileContention)
 	if err := profileDiagnostics.overrideWithEnv(); err != nil {
 		return err
 	}
 
 	if profileDiagnostics.enabled {
-		fmt.Println("diagnostics: pprof profiling enabled", "addr", profileDiagnostics.addr, "port", profileDiagnostics.port)
-		runtime.SetBlockProfileRate(1)
+		fmt.Println("diagnostics: pprof profiling enabled", "addr", profileDiagnostics.addr, "port", profileDiagnostics.port, "contentionProfilingEnabled", profileDiagnostics.contention)
+
+		if profileDiagnostics.contention {
+			runtime.SetBlockProfileRate(1)
```

Comment:
> But maybe I am overthinking and this flag is for development only.

nope this is definitely for production use as well 😅

why wouldn't 100% be a good idea for prod?

Reply:
> I am a bit surprised to see a bool flag

Something we picked up from k8s apiserver that uses a similar flag

Reply:
> why wouldn't 100% be a good idea for prod?

Increased memory usage and cpu consumption. (I don't have any specific numbers)

Here are some examples from non pyroscope:
[gcloud](https://github.com/googleapis/google-cloud-go/blob/a187451a912835703078e5b6a339c514edebe5de/profiler/mutex.go#L21): 1/100 - 1%

[datadog](https://github.com/DataDog/dd-trace-go/blob/f00b4a740e6d1aec0d60a5cce71349272dbdb299/profiler/options.go#L37): 1/10 - 10%

After all, we do not set `runtime.MemProfileRate = 1` ? :D

I guess it could be useful to set different values for different deployments, like if you have 1000k pods maybe 1% is enough and if you have 1 pod, maybe 10% could be enough. idk. I feel like If I was running my own instance I'd like to have control over it, although in practice I would be happy with anything != 1 and would not change default manually unless I observe any issues (maybe we could keep it as is and just wait).

These are just my thoughts, I am not trying to ask you to change something.

Reply:
thanks @korniltsev - that's helpful!

@marefr I think it'd be worthwhile to make the value(s) configurable, and perhaps be able to control `SetMutexProfileFraction` independently?

Reply:
Make sense 👍 

I'll work on some updates and let you know.

Reply:
I've pushed some changes. Let me know what you think